cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/sidecar/ArtifactLocalizer.java

```diff
@@ -105,6 +105,20 @@ public ArtifactLocalizer(CConfiguration cConf, RemoteClientFactory remoteClientF
     this.dataDir = cConf.get(Constants.CFG_LOCAL_DATA_DIR);
   }
 
+  /**
+   * Gets the location on the local filesystem for the given artifact. This method handles fetching the artifact as well
+   * as caching it.
+   *
+   * @param artifactId The ArtifactId of the artifact to fetch
+   * @return The Local Location for this artifact
+   * @throws ArtifactNotFoundException if the given artifact does not exist
+   * @throws IOException if there was an exception while fetching or caching the artifact
+   * @throws Exception if there was an unexpected error
+   */
+  public File getArtifact(ArtifactId artifactId) throws Exception {
+    return Retries.callWithRetries(() -> fetchArtifact(artifactId), retryStrategy);
```

Comment:
Shall we call `getArtifact` method from inside `getAndUnpackArtifact` method?

Reply:
Done